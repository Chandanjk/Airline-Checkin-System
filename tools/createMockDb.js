/* eslint-disable no-console */

const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const { login_data, flights_data, passengers_data } = mockData;
const data = JSON.stringify({ login_data, flights_data, passengers_data });
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function (err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
