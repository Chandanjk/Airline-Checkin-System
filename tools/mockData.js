const login_data = [
  {
    id: 1,
    firstName: "Chandan",
    lastName: "Kapadne",
    phone: 1234567890,
    email: "testAdmn2023@gmail.com",
    password: "admin%123",
    role: "admin",
  },
  {
    id: 2,
    firstName: "Ravi",
    lastName: "Shastri",
    phone: 9876543211,
    email: "teststaff548@gmail.com",
    password: "staff%123",
    role: "staff",
  },
  {
    id: 3,
    firstName: "Anurag",
    lastName: "Kushaq",
    phone: 9087654367,
    email: "staff_2@gmail.com",
    password: "staff2",
    role: "staff",
  },
];

const flights_data = [
  {
    id: 1,
    flightNo: 5011,
    airline: "IndiGo",
    departureStation: "BLR",
    arrivalStation: "HYD",
    departureDate: "2023-08-20T22:30:00",
    arrivalDate: "2023-08-20T23:30:00",
    special_meals: ["Veg-Thali", "Dum Biryani", "Custard", "Fruit Salad"],
    ancillary: ["Vouchers", "TV", "Goodies"],
  },
  {
    id: 2,
    flightNo: 5021,
    airline: "SpiceJet",
    departureStation: "CHE",
    arrivalStation: "MUM",
    departureDate: "2023-08-21T21:30:00",
    arrivalDate: "2023-08-21T23:30:00",
    special_meals: ["Non-veg-Thali", "Chinese Specials", "Pasta"],
    ancillary: ["Moviess", "Student Discount"],
  },
  {
    id: 3,
    flightNo: 5031,
    airline: "AirIndia",
    departureStation: "HYD",
    arrivalStation: "DEL",
    departureDate: "2023-08-22T22:40:00",
    arrivalDate: "2023-08-22T23:30:00",
    special_meals: ["Dessert", "North-Special", "Champagne"],
    ancillary: ["Goodies", "BaggageAllowance"],
  },
];

const passengers_data = [
  {
    id: 1,
    first_name: "Chandan",
    last_name: "Kapadne",
    date_of_birth: "10-Jun-1999",
    gender: "Male",
    address: "Hinjewadi Pune",
    mobile_no: 1234567890,
    seat_no: "C1",
    passport: "PASS10000",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 1,
    special_meals: ["Veg-Thali", "Dum Biryani"],
    ancillary: ["Vouchers"],
  },
  {
    id: 2,
    first_name: "Ashish",
    last_name: "Surve",
    date_of_birth: "4-Aug-1999",
    gender: "Male",
    address: "Baner Pune",
    mobile_no: 9876590456,
    seat_no: "D2",
    passport: "PASS10001",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 1,
    special_meals: ["Dum Biryanis"],
    ancillary: ["Vouchers"],
  },
  {
    id: 3,
    first_name: "Prashi",
    last_name: "Dsuza",
    date_of_birth: "15-Jul-1993",
    gender: "Female",
    address: "Navi Mumbai",
    mobile_no: 6723876545,
    seat_no: "D9",
    passport: "PASS10002",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 3,
    special_meals: ["Dessert"],
    ancillary: ["Goodies"],
  },
  {
    id: 4,
    first_name: "Shivam",
    last_name: "Samudre",
    date_of_birth: "3-Oct-1976",
    gender: "Male",
    address: "Chennai",
    mobile_no: 9898778534,
    seat_no: "B6",
    passport: "PASS10003",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 1,
    special_meals: ["Veg-Thali", "Pasta"],
    ancillary: ["TV"],
  },
  {
    id: 5,
    first_name: "AJAY",
    last_name: "Nagar",
    date_of_birth: "20-May-1999",
    gender: "Male",
    address: "Mumbai",
    mobile_no: 5743983739,
    seat_no: "",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "NC",
    flight_id: 2,
    special_meals: [],
    ancillary: [],
  },
  {
    id: 6,
    first_name: "Sachin",
    last_name: "Tendehar",
    date_of_birth: "12-Jan-1984",
    gender: "Male",
    address: "Gurugram",
    mobile_no: 6868676545,
    seat_no: "B2",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 3,
    special_meals: ["Dessert"],
    ancillary: ["BaggageAllowance"],
  },
  {
    id: 7,
    first_name: "Prakas",
    last_name: "Misra",
    date_of_birth: "3-Aug-1973",
    gender: "Male",
    address: "Kolkatta",
    mobile_no: 2353567854,
    seat_no: "E3",
    passport: "PASS10004",
    wheelChair: "Yes",
    infant: "No",
    status: "AC",
    flight_id: 1,
    special_meals: ["Dum Biryanis"],
    ancillary: ["Goodies", "Vouchers"],
  },
  {
    id: 8,
    first_name: "Vikram",
    last_name: "Rastogi",
    date_of_birth: "2-May-1969",
    gender: "Male",
    address: "",
    mobile_no: 85758948654,
    seat_no: "",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "NC",
    flight_id: 2,
    special_meals: [],
    ancillary: [],
  },
  {
    id: 9,
    first_name: "Vinay",
    last_name: "Jambhulkar",
    date_of_birth: "14-Feb-1991",
    gender: "Male",
    address: "Nashik",
    mobile_no: 6789234565,
    seat_no: "",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "NC",
    flight_id: 3,
    special_meals: ["Champagne"],
    ancillary: ["BaggageAllowance"],
  },
  {
    id: 10,
    first_name: "Raju",
    last_name: "Srivastava",
    date_of_birth: "3-Aug-1997",
    gender: "Male",
    address: "Goa",
    mobile_no: 9889756854,
    seat_no: "E6",
    passport: "PASS10005",
    wheelChair: "Yes",
    infant: "Yes",
    status: "AC",
    flight_id: 1,
    special_meals: ["Dum Biryanis"],
    ancillary: ["Vouchers", "TV"],
  },
  {
    id: 11,
    first_name: "Sumeet",
    last_name: "Gaitonde",
    date_of_birth: "20-May-1984",
    gender: "Male",
    address: "New Delhi",
    mobile_no: 8768453274,
    seat_no: "",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "NC",
    flight_id: 2,
    special_meals: [],
    ancillary: [],
  },
  {
    id: 12,
    first_name: "Raviraj",
    last_name: "Datta",
    date_of_birth: "2-Sep-1980",
    gender: "Male",
    address: "Hyderabad",
    mobile_no: 1212176545,
    seat_no: "C2",
    passport: "PASS10006",
    wheelChair: "Yes",
    infant: "No",
    status: "AC",
    flight_id: 1,
    special_meals: ["Veg-Thali"],
    ancillary: ["TV"],
  },
  {
    id: 13,
    first_name: "Prem",
    last_name: "Kumar",
    date_of_birth: "30-Aug-1987",
    gender: "Male",
    address: "Bangalore",
    mobile_no: 9893678554,
    seat_no: "C4",
    passport: "PASS10007",
    wheelChair: "Yes",
    infant: "Yes",
    status: "AC",
    flight_id: 2,
    special_meals: ["Chinese Specials", "Pasta"],
    ancillary: ["Movies", "Student Discount"],
  },
  {
    id: 14,
    first_name: "Dev",
    last_name: "Rustom",
    date_of_birth: "21-Mar-1994",
    gender: "Male",
    address: "Chennai",
    mobile_no: 8555558654,
    seat_no: "A4",
    passport: "PASS10008",
    wheelChair: "No",
    infant: "Yes",
    status: "AC",
    flight_id: 1,
    special_meals: [],
    ancillary: ["TV"],
  },
  {
    id: 15,
    first_name: "Virat",
    last_name: "Lorli",
    date_of_birth: "13-Dec-1989",
    gender: "Male",
    address: "Punjab",
    mobile_no: 6789899999,
    seat_no: "",
    passport: "",
    wheelChair: "No",
    infant: "No",
    status: "NC",
    flight_id: 3,
    special_meals: ["Champagne"],
    ancillary: ["BaggageAllowance"],
  },
  {
    id: 16,
    first_name: "Mukesh",
    last_name: "Topiwala",
    date_of_birth: "3-Aug-2000",
    gender: "Male",
    address: "Bengal",
    mobile_no: 3333367854,
    seat_no: "B1",
    passport: "PASS10009",
    wheelChair: "Yes",
    infant: "Yes",
    status: "AC",
    flight_id: 1,
    special_meals: ["Veg-Thali", "Dum Biryanis"],
    ancillary: ["TV"],
  },
  {
    id: 17,
    first_name: "Raghuram",
    last_name: "Mishra",
    date_of_birth: "10-May-1982",
    gender: "Male",
    address: "Pune",
    mobile_no: 8768934354,
    seat_no: "B1",
    passport: "PASS10010",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 2,
    special_meals: ["Non-veg-Thali"],
    ancillary: ["Movies"],
  },
  {
    id: 18,
    first_name: "Akash",
    last_name: "Bhuse",
    date_of_birth: "1-Jan-1991",
    gender: "Male",
    address: "Vadodra",
    mobile_no: 6789888885,
    seat_no: "B3",
    passport: "PASS10011",
    wheelChair: "No",
    infant: "No",
    status: "AC",
    flight_id: 3,
    special_meals: ["North-Special"],
    ancillary: ["BaggageAllowance"],
  },
  {
    id: 19,
    first_name: "Megha",
    last_name: "Kulkarni",
    mobile_no: 8963434383,
    address: "Malegaon",
    gender: "Female",
    infant: "Yes",
    wheelChair: "No",
    flight_id: 2,
    status: "NC",
    seat_no: "",
    passport: "PASS10012",
    date_of_birth: "",
    special_meals: ["Pasta"],
    ancillary: ["Movies"],
  },
  {
    id: 20,
    first_name: "Rakesh",
    last_name: "Roshan",
    mobile_no: 9090333330,
    address: "Dhule",
    gender: "Male",
    infant: "No",
    wheelChair: "No",
    flight_id: 1,
    status: "AC",
    seat_no: "B4",
    passport: "PASS10013",
    date_of_birth: "15-feb-1998",
    special_meals: ["Veg-Thali"],
    ancillary: ["Goodies", "Vouchers"],
  },
  {
    id: 21,
    first_name: "Shyam",
    last_name: "Murali",
    mobile_no: 9097777728,
    address: "Bangalore",
    gender: "Male",
    infant: "Yes",
    wheelChair: "No",
    flight_id: 1,
    status: "AC",
    seat_no: "B7",
    passport: "PASS10014",
    date_of_birth: "28-Feb-2001",
    special_meals: ["Veg-Thali"],
    ancillary: ["TV"],
  },
  {
    id: 22,
    first_name: "Tony",
    last_name: "Satark",
    mobile_no: 9232333849,
    address: "New York",
    gender: "Male",
    infant: "No",
    wheelChair: "No",
    flight_id: 1,
    status: "AC",
    seat_no: "C4",
    passport: "",
    date_of_birth: "",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 23,
    first_name: "Llyoid",
    last_name: "Forger",
    mobile_no: 7777777777,
    address: "UK",
    gender: "Male",
    infant: "No",
    wheelChair: "Yes",
    flight_id: 1,
    status: "AC",
    seat_no: "C7",
    passport: "",
    date_of_birth: "",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 24,
    first_name: "Genya",
    last_name: "Mochizuki",
    mobile_no: 1111111111,
    address: "Japan",
    gender: "Male",
    infant: "Yes",
    wheelChair: "No",
    flight_id: 1,
    status: "NC",
    seat_no: "",
    passport: "",
    date_of_birth: "",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 25,
    first_name: "Chitralekha",
    last_name: "Pavnekar",
    mobile_no: 9898985598,
    address: "Delhi",
    gender: "Female",
    infant: "No",
    wheelChair: "No",
    flight_id: 3,
    status: "NC",
    seat_no: "",
    passport: "",
    date_of_birth: "",
    special_meals: ["Champagne"],
    ancillary: [],
  },
  {
    id: 26,
    first_name: "Sanju",
    last_name: "Datt",
    mobile_no: 2222221712,
    address: "South Bombay",
    gender: "Male",
    infant: "Yes",
    wheelChair: "No",
    flight_id: 1,
    status: "AC",
    seat_no: "E1",
    passport: "",
    date_of_birth: "07-jun-1976",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 27,
    first_name: "Anya",
    last_name: "Forger",
    mobile_no: 7344438383,
    address: "Tokyo",
    gender: "Female",
    infant: "Yes",
    wheelChair: "Yes",
    flight_id: 1,
    status: "NC",
    seat_no: "",
    passport: "",
    date_of_birth: "",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 28,
    first_name: "Eebu",
    last_name: "Hatela",
    mobile_no: 4222223534,
    address: "Hyderabad",
    gender: "Male",
    infant: "Yes",
    wheelChair: "No",
    flight_id: 1,
    status: "NC",
    seat_no: "",
    passport: "",
    date_of_birth: "31-Dec-1976",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 29,
    first_name: "Deepak",
    last_name: "Kalal",
    mobile_no: 9999998989,
    gender: "Male",
    infant: "Yes",
    wheelChair: "Yes",
    flight_id: 2,
    status: "NC",
    seat_no: "",
    country: "",
    date_of_birth: "20-Dec-1980",
    passport: "PASS10067",
    address: "Jammu",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 30,
    first_name: "Naruto",
    last_name: "Uzumaki",
    mobile_no: 1111111111,
    gender: "Male",
    infant: "No",
    wheelChair: "No",
    flight_id: 3,
    status: "NC",
    seat_no: "",
    date_of_birth: "01-Jan-1997",
    passport: "PASS10087",
    address: "Konoha",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 31,
    first_name: "Hinata",
    last_name: "Hydo",
    mobile_no: 4545488989,
    gender: "Female",
    infant: "No",
    wheelChair: "No",
    flight_id: 1,
    status: "NC",
    seat_no: "",
    date_of_birth: "",
    passport: "",
    address: "",
    special_meals: [],
    ancillary: [],
  },
  {
    id: 32,
    first_name: "Sasuke",
    last_name: "Uchiha",
    mobile_no: 3555553434,
    gender: "Male",
    infant: "No",
    wheelChair: "No",
    flight_id: 1,
    status: "NC",
    seat_no: "",
    date_of_birth: "29-Nov-1985",
    passport: "",
    address: "",
    special_meals: [],
    ancillary: [],
  },
];

module.exports = {
  login_data,
  flights_data,
  passengers_data,
};
