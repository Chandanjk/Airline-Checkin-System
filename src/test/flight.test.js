import React from "react";
import FlightDetails from "../components/flights/flightDetails";
import { flights_data } from "../../tools/mockData";
import * as renderer from "react-test-renderer";

it("Renders flight details", () => {
  const tree = renderer.create(
    <FlightDetails flightDetails={flights_data[0]} />
  );
  expect(tree).toMatchSnapshot();
});
