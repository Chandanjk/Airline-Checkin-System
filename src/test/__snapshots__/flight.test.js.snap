// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders flight details 1`] = `
<div
  className="card"
  style={
    {
      "marginTop": "15px",
    }
  }
>
  <h3
    className="card-header text-center font-weight-bold text-uppercase py-4 tempting-azure-gradient"
  >
    Flight Details
  </h3>
  <div
    className="card-body frozen-dreams-gradient"
  >
    <div
      className="container"
      style={
        {
          "borderStyle": "solid",
          "padding": "5px",
        }
      }
    >
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <b>
            AIRLINE : 
          </b>
           
          IndiGo
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            FLIGHT No : 
          </b>
           
          5011
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            DEP STN : 
          </b>
           
          BLR
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            ARV STN : 
          </b>
           
          HYD
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            DEP DATE : 
          </b>
           
          2023-08-20
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            ARV DATE : 
          </b>
           
          2023-08-20
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            DEP TIME : 
          </b>
           
          22:30
        </div>
        <div
          className="col-sm-6"
        >
          <b>
            ARV TIME : 
          </b>
           
          23:30
        </div>
      </div>
    </div>
  </div>
</div>
`;
